internal/service/storage/storage_backup.go

```diff
@@ -0,0 +1,227 @@
+package storage
+
+import (
+	"context"
+	"time"
+
+	"github.com/UpCloudLtd/terraform-provider-upcloud/internal/utils"
+	"github.com/UpCloudLtd/upcloud-go-api/v8/upcloud"
+	"github.com/UpCloudLtd/upcloud-go-api/v8/upcloud/request"
+	"github.com/UpCloudLtd/upcloud-go-api/v8/upcloud/service"
+	"github.com/hashicorp/terraform-plugin-framework/resource"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/planmodifier"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/stringplanmodifier"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+)
+
+var (
+	_ resource.Resource              = &storageBackupResource{}
+	_ resource.ResourceWithConfigure = &storageBackupResource{}
+)
+
+type storageBackupResource struct {
+	client *service.Service
+}
+
+func NewStorageBackupResource() resource.Resource {
+	return &storageBackupResource{}
+}
+
+func (r *storageBackupResource) Metadata(_ context.Context, req resource.MetadataRequest, resp *resource.MetadataResponse) {
+	resp.TypeName = req.ProviderTypeName + "_storage_backup"
+}
+
+// Configure adds the provider configured client to the resource.
+func (r *storageBackupResource) Configure(_ context.Context, req resource.ConfigureRequest, resp *resource.ConfigureResponse) {
+	r.client, resp.Diagnostics = utils.GetClientFromProviderData(req.ProviderData)
+}
+
+type storageBackupModel struct {
+	SourceStorage types.String `tfsdk:"source_storage"`
+	Title         types.String `tfsdk:"title"`
+	ID            types.String `tfsdk:"id"`
+	CreatedAt     types.String `tfsdk:"created_at"`
+}
+
+func (r *storageBackupResource) Schema(_ context.Context, _ resource.SchemaRequest, resp *resource.SchemaResponse) {
+	resp.Schema = schema.Schema{
+		Description: "Manages an on-demand storage backup.",
+		Attributes: map[string]schema.Attribute{
+			"source_storage": schema.StringAttribute{
+				Required:    true,
+				Description: "The UUID of the storage to back up.",
+			},
```

Comment:
Could use `RequiresReplace` plan modifier here (similarly than in template resource):

```suggestion
				PlanModifiers: []planmodifier.String{
					stringplanmodifier.RequiresReplace(),
				},
			},
```

Reply:
Like it